components/raftstore/src/store/msg.rs

```diff
@@ -320,31 +321,46 @@ impl<EK: KvEngine> fmt::Debug for CasualMessage<EK> {
 
 /// Raft command is the command that is expected to be proposed by the
 /// leader of the target raft group.
-#[derive(Debug)]
 pub struct RaftCommand<S: Snapshot> {
     pub send_time: Instant,
     pub request: RaftCmdRequest,
     pub callback: Callback<S>,
+    pub pw_callback: Option<UtilCallback<()>>,
```

Comment:
How about moving it to `Callback`? For example,
```rust
pub enum Callback<S: Snapshot> {
    ...
    /// Write callback.
    Write {
        cb: WriteCallback,
        /// `post_propose` is called after a request is proposed to the raft group successfully.
        /// It's used to notify the caller to move on early because it's very likely the request
        /// will be applied to the raftstore.
        post_propose: Option<UtilCallback<()>>,
    },
}
```
or adding a new variant
```rust
pub enum Callback<S: Snapshot> {
    ...
    /// Write callback.
    Write(WriteCallback)
    /// Write callback with extensions.
    WriteExt {
        cb: WriteCallback,
        /// `post_propose` is called after a request is proposed to the raft group successfully.
        /// It's used to notify the caller to move on early because it's very likely the request
        /// will be applied to the raftstore.
        post_propose: Option<UtilCallback<()>>,
    },
}
```

I think it can simplify codes.